theseus/labs/lie/lie_tensor.py

```diff
@@ -32,35 +57,115 @@ def _get_fn_lib(ltype: _ltype):
     }[ltype]
 
 
-class LieTensor:
-    def __init__(self, data_tensor: torch.Tensor, ltype: _ltype):
-        self._t = data_tensor
-        self._fn_lib = _get_fn_lib(ltype)
-        self._fn_lib.check_group_tensor(data_tensor)
+class _LieTensorBase(torch.Tensor):
+    def __new__(cls, data: Any, ltype: _ltype, requires_grad=None):
+        if requires_grad is None:
```

Comment:
I would not have this case and just the one with `_make_subclass` below. It will make your autograd story simpler:
`lt = _LieTensorBase(t)` is never a differentiable op (always creates a leaf). And so lt never has an autograd graph. You can use `requires_grad` flag just like `torch.Tensor()` though to make it require grad if you want.

If you want a way to create your class in a differentiable way, use a custom method implemented with a custom autograd Function such as `lt = _LieTensorBase.from_tensor(t)` which will always be differentiable.